controllers/machine_controller.go

```diff
@@ -181,6 +182,42 @@ func (r *MachineReconciler) reconcile(ctx context.Context, cluster *clusterv1.Cl
 	return res, kerrors.NewAggregate(errs)
 }
 
+func (r *MachineReconciler) reconcileLabels(ctx context.Context, m *clusterv1.Machine) error {
+	logger := r.Log.WithValues("machine", m.Name, "namespace", m.Namespace)
+
+	if m.Labels == nil {
+		m.Labels = make(map[string]string)
+	}
+
+	// return early if machine already has both the labels for machineSet and machineDeployment controller
+	if m.Labels[clusterv1.MachineDeploymentLabelName] != "" && m.Labels[clusterv1.MachineSetLabelName] != "" {
```

Comment:
```suggestion
	if m.Labels[clusterv1.MachineDeploymentLabelName] != "" || m.Labels[clusterv1.MachineSetLabelName] != "" {
```

If the Machine is in a MachineSet but not in a MachineDeployment, we'll end up reconciling it all the time

Reply:
Updated as you suggested, but I was thinking if user somehow removes one of the labels then we need to reconcile again